### PR TITLE
 PP-10343 Add progress indicator for tasklist cypress tests

### PR DESCRIPTION
**Pull request by AbdirizakIdris**
- Added logic in nunjucks file to show progress indicator when ENABLE_STRIPE_ONBOARDING_TASK_LIST is true
- Updated cypress test to check progress of tasks

### CODE REVIEW

File: app/views/your-psp/_stripe.njk [hunk 1]
```diff
@@ -94,6 +94,12 @@
  {% else %}
  {% if  enableStripeOnboardingTaskList %}
     <h1 class="govuk-heading-m govuk-!-margin-top-8">Information for Stripe</h1>
+    <h3 class="govuk-body"> {{progressIndicator.no_of_tasks_completed}} out of {{progressIndicator.total_number_of_tasks}} steps completed</h3>
```
Comment (stephencdaly) [inline app/views/your-psp/_stripe.njk:97]: I don't think this should be a h3.
Are we following the guidance in the task list pattern? https://design-system.service.gov.uk/patterns/task-list-pages/

If so, looks like the progress indicator should be below the h2 in a `<p> `- as in this example https://prototype-kit.service.gov.uk/v12/docs/templates/task-list

**Review by stephencdaly — COMMENTED** [review]
Spotted one thing

File: app/views/your-psp/_stripe.njk (hunk 1 above)
Comment (AbdirizakIdris) [reply app/views/your-psp/_stripe.njk:97]: You're right, good spot

File: app/views/your-psp/_stripe.njk [hunk 2]
```diff
@@ -91,57 +92,67 @@
     </div>
   </div>
 
- {% else %}
- {% if  enableStripeOnboardingTaskList %}
+{% else %}
+  {% if enableStripeOnboardingTaskList %}
     <h1 class="govuk-heading-m govuk-!-margin-top-8">Information for Stripe</h1>
```
Comment (stephencdaly) [inline app/views/your-psp/_stripe.njk:97]: We already have a `h1` on this page, and we should only have 1. If we're keeping this header, it should be a h2 and the current h2s should become h3.

File: app/views/your-psp/_stripe.njk [hunk 3]
```diff
@@ -91,57 +92,67 @@
...
+
+    {% if progressIndicator.no_of_tasks_completed < progressIndicator.total_number_of_tasks %}
+      <h2 class="govuk-heading-m govuk-!-margin-top-8">Information incomplete</h2>
```
Comment (stephencdaly) [inline app/views/your-psp/_stripe.njk:100]: Screenshot 2023-02-28 at 17 05 15

The styling doesn't look quite right here. The h2 should have the following classes `<h2 class="govuk-heading-s govuk-!-margin-bottom-2">` according to the design system guidance.

Are we working from a mocked up design from this page? Is it worth checking with Johnny whether we want all the headers? And if so what styling they should have?

**Review by stephencdaly — APPROVED** [review]
LGTM! The two h1s on the page will be addressed in a different PR.

**Pull request merged**